react/Provider/provider.tsx

```diff
@@ -9,13 +13,110 @@ const Provider: FC = props => {
   const [html, setHtml] = useState('')
   const [file, setFile] = useState({ files: null, result: null })
   const [text, setText] = useState('')
+  const [conditions, setConditions] = useState({
+    simpleStatements: [],
+    operator: 'all',
+  })
+
+  const [showAlert, setShowAlert] = useState(0)
+
+  const [textValidate, setTextValidate] = useState<string[]>([''])
+  const [saveMutation] = useMutation(uploadFile)
+  const [saveMasterdataMutation] = useMutation(saveMasterdata)
+
+  const handleCloseAlert = () => {
+    setShowAlert(0)
+  }
 
   function chooseFile(files: any) {
     setFile({ ...file, ...{ result: files } })
   }
 
-  function save() {
-    // inserir regra
+  function validateIfAllFieldsIsComplete() {
+    const validation = []
+
+    if (name === null || name === undefined || name === '') {
```

Comment:
You can do this here:

```suggestion
    if (!name) {
```

Reply:
Similar [here](https://github.com/vtex-apps/badges/pull/4/files#diff-1d25edb2b7119b3ab49f6e24d11a2a123b475183c2f82435efda2bf7fb1f225dR46) and [here](https://github.com/vtex-apps/badges/pull/4/files#diff-1d25edb2b7119b3ab49f6e24d11a2a123b475183c2f82435efda2bf7fb1f225dR50).